endpoints/openrtb2/amp_auction.go

```diff
@@ -262,6 +263,21 @@ func (deps *endpointDeps) AmpAuction(w http.ResponseWriter, r *http.Request, _ h
 	}
 }
 
+// Sets the effective publisher ID for an AMP request
+func setEffectiveAmpPubID(req *openrtb.BidRequest, urlValues url.Values) {
+	// Try to look for the pub ID in the ususal places first
+	setEffectivePubID(req)
+
+	// For amp requests, the publisher ID could be sent via the account
+	// query string
+	if req.Site.Publisher.ID == pbsmetrics.PublisherUnknown {
+		if acc := urlValues.Get("account"); acc != "" {
```

Comment:
Can you please add a check to ignore unresolved macros? If the account is resolved to "ACCOUNT_ID" then it's unresolved.

Reply:
Could there be other such unresolved macros? And is there a way for us to know what they might be? 

Also, since PBS is pretty much just a channel to pass this value, I am wondering if this is the kind of validation that should be done either downstream in the system that is calling us or upstream in the bidders where they actually use account ID value.

Reply:
There might be other unresolved macros. Not sure. This is one we've seen in the real world though and is an annoyance in the metrics.